chain-extensions/types/assets/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use scale_info::scale;
+use sp_runtime::{DispatchError, ModuleError};
+
+#[derive(PartialEq, Eq, Copy, Clone, scale::Encode, scale::Decode, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum AssetsError {
+    /// Success
+    Success = 0,
+    /// Account balance must be greater than or equal to the transfer amount.
+    BalanceLow = 1,
+    /// The account to alter does not exist.
+    NoAccount = 2,
+    /// The signing account has no permission to do the operation.
+    NoPermission = 3,
+    /// The given asset ID is unknown.
+    Unknown = 4,
+    /// The origin account is frozen.
+    Frozen = 5,
+    /// The asset ID is already taken.
+    InUse = 6,
+    /// Invalid witness data given.
+    BadWitness = 7,
+    /// Minimum balance should be non-zero.
+    MinBalanceZero = 8,
+    /// Unable to increment the consumer reference counters on the account. Either no provider
+    /// reference exists to allow a non-zero balance of a non-self-sufficient asset, or the
+    /// maximum number of consumers has been reached.
+    NoProvider = 9,
+    /// Invalid metadata given.
+    BadMetadata = 10,
+    /// No approval exists that would allow the transfer.
+    Unapproved = 11,
+    /// The source account would not survive the transfer and it needs to stay alive.
+    WouldDie = 12,
+    /// The asset-account already exists.
+    AlreadyExists = 13,
+    /// The asset-account doesn't have an associated deposit.
+    NoDeposit = 14,
+    /// The operation would result in funds being burned.
+    WouldBurn = 15,
+    /// Unknown error
+    RuntimeError = 99,
+}
+
+impl TryFrom<DispatchError> for AssetsError {
+    type Error = DispatchError;
+
+    fn try_from(input: DispatchError) -> Result<Self, Self::Error> {
+        let error_text = match input {
+            DispatchError::Module(ModuleError { message, .. }) => message,
+            _ => Some("No module error Info"),
+        };
+        return match error_text {
```

Comment:
Not related to this PR in particular, but it might be good to have some integration test to ensure change in pallet doesn't break the CE. Right now only a string, with no bounds to `pallet-assets` `Error` enum exists.

Or perhaps this could be changed to something like `pallet_assets::Error::<T>::BalanceLow.to_string()`?
That way we can rely on compiler to detect if some enum value is renamed.